From 37a407b5524675a03966b380eea89809ad10d86d Mon Sep 17 00:00:00 2001
From: Hadrien LEMAIRE <had.lemaire@gmail.com>
Date: Fri, 12 Feb 2021 11:02:50 +0100
Subject: [PATCH 6/8] scrollback-mouse-increment

---
 config.def.h | 5 +++--
 1 file changed, 3 insertions(+), 2 deletions(-)

diff --git a/config.def.h b/config.def.h
index d29a673..f58e7d3 100644
--- a/config.def.h
+++ b/config.def.h
@@ -172,10 +172,11 @@ static uint forcemousemod = ShiftMask;
  * Internal mouse shortcuts.
  * Beware that overloading Button1 will disable the selection.
  */
+const unsigned int mousescrollincrement = 1;
 static MouseShortcut mshortcuts[] = {
 	/* mask                 button   function        argument       release */
-	{ ShiftMask,            Button4, kscrollup,      {.i = 1} },
-	{ ShiftMask,            Button5, kscrolldown,    {.i = 1} },
+	{ Button4,              ShiftMask, kscrollup,    {.i =  mousescrollincrement} },
+	{ Button5,              ShiftMask, kscrolldown,  {.i =  mousescrollincrement} },
 	{ XK_ANY_MOD,           Button2, selpaste,       {.i = 0},      1 },
 	{ ShiftMask,            Button4, ttysend,        {.s = "\033[5;2~"} },
 	{ XK_ANY_MOD,           Button4, ttysend,        {.s = "\031"} },
-- 
2.30.0

